src/commands/createFunction/createFunction.ts

```diff
@@ -49,20 +49,32 @@ export async function createFunctionFromCommand(
 export async function createFunctionInternal(context: IActionContext, options: api.ICreateFunctionOptions): Promise<void> {
     let workspaceFolder: WorkspaceFolder | undefined;
     let workspacePath: string | undefined = options.folderPath;
+
     if (workspacePath === undefined) {
-        workspaceFolder = await getWorkspaceFolder(context, options);
-        workspacePath = workspaceFolder.uri.fsPath;
+        workspaceFolder = await getWorkspaceFolder();
+        workspacePath = workspaceFolder?.uri.fsPath;
     } else {
         workspaceFolder = getContainingWorkspace(workspacePath);
     }
 
-    addLocalFuncTelemetry(context, workspacePath);
-
-    const projectPath: string | undefined = await verifyAndPromptToCreateProject(context, workspaceFolder || workspacePath, options);
+    let projectPath: string | undefined = await verifyProjectPath(context, workspaceFolder || workspacePath);
     if (!projectPath) {
-        return;
+        // If we cannot find a valid Functions project, we need to put the user into the 'Create New Project' flow..
+        context.telemetry.properties.noWorkspaceResult = 'createNewProject';
+        await createNewProjectInternal(context, options);
+
+        if (!workspace.workspaceFolders || workspace.workspaceFolders.length === 0) {
```

Comment:
Part of creating a new project is creating a new function, so it should probably just return here.